discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -15,67 +15,75 @@
 use std::str::FromStr;
 
 use async_trait::async_trait;
+use chrono::{NaiveDateTime, Utc};
 use displaydoc::Display;
+use num_traits::FromPrimitive;
 use sqlx::{
     sqlite::{Sqlite, SqliteConnectOptions, SqlitePoolOptions},
     Pool,
+    QueryBuilder,
 };
 use thiserror::Error;
 use url::Url;
 use uuid::Uuid;
 use xayn_discovery_engine_ai::GenericError;
+use xayn_discovery_engine_providers::Market;
 
-use crate::document::{HistoricDocument, Id};
+use crate::document::{self, HistoricDocument};
 
-use super::Storage;
+use super::{
+    models::{ApiDocumentView, NewsResource, NewscatcherData},
+    FeedScope,
+    Storage,
+};
 
 #[derive(Error, Debug, Display)]
-pub enum Error {
-    /// Failed to initialize database: {0}
-    Init(#[source] sqlx::Error),
-
-    /// Failed to acquire connection: {0}
-    AcquireConnection(#[source] sqlx::Error),
-
+pub enum DatabaseError {
     /// Failed to migrate database: {0}
     Migration(#[from] sqlx::migrate::MigrateError),
 
-    /// Failed to fetch: {0}
-    Fetch(#[source] sqlx::Error),
+    /// Sql error: {0}
+    Sql(#[source] sqlx::Error),
 
     /// Failed to covert type: {0}
     TypeConversion(#[source] GenericError),
 }
 
+#[derive(Error, Debug, Display)]
+pub enum StorageError {
+    /// Database error: {0}
+    Database(#[from] DatabaseError),
+}
+
 #[derive(Clone)]
 pub(crate) struct SqliteStorage {
     pool: Pool<Sqlite>,
 }
 
 impl SqliteStorage {
-    pub(crate) async fn connect(uri: &str) -> Result<Self, Error> {
+    pub(crate) async fn connect(uri: &str) -> Result<Self, StorageError> {
         let opt = SqliteConnectOptions::from_str(uri)
-            .map_err(Error::Init)?
+            .map_err(DatabaseError::Sql)?
             .create_if_missing(true);
 
         let pool = SqlitePoolOptions::new()
             .connect_with(opt)
             .await
-            .map_err(Error::Init)?;
+            .map_err(DatabaseError::Sql)?;
 
```

Comment:
do we need to enable foreign key constraints or is that a default for `SqliteConnectionOptions` (it isn't a default for sqlite3 by itself).

Reply:
it is enabled by [default](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.foreign_keys) but good point i forgot to check that